### PR TITLE
AEIM-2116 - Open 6443 on haproxy servers

### PR DESCRIPTION
**Pull request by daaang**
Not the best way to do it maybe, but this will get it done, and it'll be easy enough to revisit once we have something working.

### CODE REVIEW

File: manifests/profile/vmhost/prereqs.pp [hunk 1]
```diff
@@ -34,7 +34,7 @@
   package { 'qemu':
     ensure => 'installed'
   }
-  
+
   package { 'qemu-kvm':
```
Comment (daaang) [inline manifests/profile/vmhost/prereqs.pp]: This was just to make the linter happy.

File: manifests/profile/haproxy.pp [hunk 2]
```diff
@@ -114,4 +114,6 @@
     tag    => 'haproxy'
   }
 
+  nebula::haproxy::kubernetes_cluster { 'just open port 6443 for now': }
```
Comment (aelkiss) [inline manifests/profile/haproxy.pp]: Why not just put the firewall rules right here?

Comment (daaang) [reply manifests/profile/haproxy.pp]: My thinking was that we'd need a defined type eventually, so I just went ahead and made one—but looking back I think that for now it'd be better to do this the simpler way and just put the firewall rules here, yeah.

**Comment by aelkiss** [comment]
What's the motivation for using virtual resources here? It seems confusing to have the firewall rules for kubernetes in a profile called 'ssh.pp' -- why can't we just put the firewall resource directly in `manifests/profile/haproxy.pp`?

**Comment by daaang** [comment]
My motivation for putting it there was to have `nebula::profile::networking::firewall::ssh` be the only thing that knows about `nebula::profile::networking::firewall::ssh::networks`, rather than have the haproxy profile loop through `lookup('nebula::profile::networking::firewall::ssh::networks')`. This way, the ssh profile still bears the responsibility of knowing which CIDRs might want access, and then it's up to haproxy and kubernetes to decide to actually listen.

There's probably a better way to do it, but I think it'll necessitate structural changes to the way we open ports for ssh access as well, and that feels out of scope to me.

**Comment by aelkiss** [comment]
I see - let's go with this for now and revisit later for the firewall stuff.

**Pull request merged**